osquery/core/system.cpp

```diff
@@ -564,11 +568,12 @@ DropPrivileges::~DropPrivileges() {
 #endif
 
 Status setThreadName(const std::string& name) {
-  int return_code;
 #if defined(__APPLE__)
-  return_code = pthread_setname_np(name.c_str());
+  int return_code = pthread_setname_np(name.c_str());
+  return Status(return_code == 0 ? 0 : 1);
```

Comment:
Would you like to make it explicit? Smth like:
```c++
return return_code == 0 ? Status::success() : Status::failure();
```

Reply:
Sure. Thanks for pointing out